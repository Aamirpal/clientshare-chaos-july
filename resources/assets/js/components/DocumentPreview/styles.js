export const styles = {
  mainHeader: {
    padding: 0,
  },
  modalHeader: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  smallHeader: {
    width: '95%',
  },
  largeHeader: {
    width: '97.2%',
  },
  smallModal: {
    minWidth: 587,
  },
  largeModal: {
    maxWidth: 1280,
    width: '94%',
    '@media (max-width: 767px)': {
      minWidth: '100%',
    },
  },
  titleContainer: {
    width: '50%',
  },
  fileName: {
    margin: 0,
    color: ({ theme }) => theme.basic_color,
    lineHeight: '21px',
    width: '100%',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    display: 'inline-block',
    whiteSpace: 'nowrap',
  },
  rightButtons: {
    display: 'flex',
  },
  buttonContainer: {
    position: ({ theme }) => theme.relative,
    display: 'flex',
    padding: '0 20px 0 15px',
    '&:after': {
      position: ({ theme }) => theme.absolute,
      content: '""',
      background: ({ theme }) => theme.dark_white,
      width: 1,
      height: 54,
      top: -18,
      right: 0,
      bottom: 0,
    },
  },
  buttonStyle: {
    color: ({ theme }) => theme.primary_color,
    margin: '0 7px 0 0',
    lineHeight: '19px',
    cursor: 'pointer',
  },
  post_image_load: {
    height: 500,
    background: ({ theme }) => theme.light_gray,
    borderRadius: 4,
    '@media (max-width: 767px)': {
      height: 300,
    },
  },
  contentContainer: {
    padding: 17,
  },
  imgContainer: {
    height: 'auto',
    borderRadius: 8,
    width: 'auto',
    maxWidth: '100%',
  },
};
