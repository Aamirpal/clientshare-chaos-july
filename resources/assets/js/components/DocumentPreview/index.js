import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';
import get from 'lodash/get';
import cx from 'classnames';

import MediaQuery from 'react-responsive';
import {
  Modal, Heading, Icon, Image, Spinner,
} from '../index';
import withTheme from '../../utils/hoc/withTheme';
import { getType, downloadFile, getFileExtension } from '../../utils/methods';
import { getUrlPreviewApi, addView } from '../../api/app';
import { downloadIcon, expandIcon, minimizeIcon } from '../../images';
import { styles } from './styles';

const DocumentPreview = React.memo(({ modelProps, file, classes }) => {
  const [size, setSize] = useState('largeModal');
  const [docData, setDocData] = useState(null);
  const metadata = get(file, 'metadata', {});
  const fileExtension = get(file, 'metadata.extention', null) || getFileExtension(get(file, 'metadata.url', ''));
  const fileType = getType(fileExtension);

  useEffect(() => {
    if (fileType === 'files') {
      getUrlPreviewApi({ url: metadata.url, extension: fileExtension }).then(({ data }) => {
        setDocData(fileExtension.toLowerCase() === 'pdf' ? get(data, 'pdf', null) : get(data, 'doc', null));
      // Implementation
      }).catch(() => false);
    }
    addView(get(file, 'post_id', null)).catch(() => false);
  }, file);

  const toggleSize = () => setSize(previous => (previous === 'smallModal' ? 'largeModal' : 'smallModal'));

  const renderImage = () => (
    <Image position="center" img={get(file, 'post_file_url', null) || get(file, 'file_url', null)} round={false} size="auto" loadingClass={classes.post_image_load} extraClass={classes.imgContainer} />
  );

  const renderVideo = () => (
    // eslint-disable-next-line jsx-a11y/media-has-caption
    <video width="100%" controls>
      <source src={get(file, 'post_file_url', null) || get(file, 'file_url', null)} type={metadata.mimeType} />
                  Your browser does not support the video tag.
    </video>
  );

  const renderViewer = () => (
    // eslint-disable-next-line jsx-a11y/iframe-has-title
    <iframe src={docData} width="100%" height="670" />
  );
  return (
    <Modal
      modelProps={{ ...modelProps, dialogClassName: classes[[size]], className: 'document-preview-popup' }}
      headerText={(
        <div className={cx(classes.modalHeader, {
          [classes.smallHeader]: size === 'smallModal',
          [classes.largeHeader]: size === 'largeModal',
        })}
        >
          <div className={classes.titleContainer}>
            <Heading as="h3" headingProps={{ className: classes.fileName }}>{metadata.originalName}</Heading>
          </div>
          <MediaQuery query="(min-device-width: 767px)">
            <div className={classes.rightButtons}>
              <div className={classes.buttonContainer} onClick={() => downloadFile(get(file, 'metadata.url', ''), get(file, 'metadata.originalName', ''))}>
                <Heading as="h4" headingProps={{ className: classes.buttonStyle }}>Download</Heading>
                <Icon path={downloadIcon} />
              </div>
              <div className={classes.buttonContainer} onClick={toggleSize}>
                <Heading as="h4" headingProps={{ className: classes.buttonStyle }}>
                  {size === 'largeModal' ? 'Minimise' : 'Expand'}
                </Heading>
                <Icon path={size === 'largeModal' ? minimizeIcon : expandIcon} />
              </div>
            </div>
          </MediaQuery>
        </div>
)}
      customHeader
      headerClass={classes.mainHeader}
    >
      <div>
        <div className={classes.contentContainer}>

          <MediaQuery query="(max-device-width: 767px)">
            <div className={cx(classes.buttonContainer, 'file-download')} onClick={() => downloadFile(get(file, 'metadata.url', ''), get(file, 'metadata.originalName', ''))}>
              <Heading as="h4" headingProps={{ className: classes.buttonStyle }}>Download</Heading>
              <Icon path={downloadIcon} />
            </div>
          </MediaQuery>

          {fileType === 'images' && renderImage()}
          {fileType === 'videos' && renderVideo()}
          {(fileType === 'files' && docData) && renderViewer()}
        </div>
        {(fileType === 'files' && !docData) && <Spinner />}
      </div>
    </Modal>
  );
});

DocumentPreview.propTypes = {
  modelProps: PropTypes.object,
};

DocumentPreview.defaultProps = {
  modelProps: {},
};
export default withTheme(injectSheet(styles)(DocumentPreview));
