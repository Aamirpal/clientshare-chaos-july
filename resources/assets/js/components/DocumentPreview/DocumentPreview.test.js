import React from 'react';
import ReactDOM from 'react-dom';
import DocumentPreview from './index';

describe('Post Feed Testing', () => {
  it('renders without crashing Post div', () => {
    const div = document.createElement('div');
    const comp = <DocumentPreview />;
    ReactDOM.render(<DocumentPreview />, div);
    ReactDOM.unmountComponentAtNode(div);
    expect(comp).toMatchSnapshot();
  });
});
