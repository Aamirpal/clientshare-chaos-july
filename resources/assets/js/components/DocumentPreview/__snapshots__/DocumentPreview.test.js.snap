// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Feed Testing renders without crashing Post div 1`] = `<Theme />`;
